templates/tapestry.js

```diff
@@ -1052,12 +1117,94 @@ function tapestryTool(config){
                     if (config.wpCanEditTapestry || d.accessible) {
-                        tapestry.selectNode(d.id)
+                        if (!isMultiSelect) {
+                            selection.clear();
+                            tapestry.selectNode(d.id);
+                        } else {
+                            if (!selection.size()) {
+                                selection.add(getNodeById(root))
+                            }
+                            selection.add(d);
+                        }
                     }
                 }
             });
     }
 
+    function dragstarted(d) {
+        if(!config.wpCanEditTapestry &&
+            tapestry.dataset.settings.nodeDraggable === false) {
+            return;
+        }
+ 
+        if (!d3.event.active) simulation.alphaTarget(0.2).restart();
+
+        if (!selection.size()) {
+            selection.add(d)
+        }
+        if (!selection.has(d) && !isMultiSelect) {
+            selection.clear();
+            selection.add(d);
+        }
+        nodesBeforeDrag = Array.from(selection)
+        recordAnalyticsEvent('user', 'drag-start', 'node', d.id, {'x': d.x, 'y': d.y});
+    }
+
+    function dragged(d) {
+        if(!config.wpCanEditTapestry &&
+            tapestry.dataset.settings.nodeDraggable === false) {
+            return;
+        }
+
+        const xBeforeDrag = d[xORfx]
+        const yBeforeDrag = d[yORfy]
+        const deltaX = d3.event.x - xBeforeDrag
+        const deltaY = d3.event.y - yBeforeDrag
+        selection.forEach(nd => {
+            if (canEditNode(nd)) {
+                nd[xORfx] = getBoundedCoord(nd[xORfx] + deltaX, tapestryDimensionsBeforeDrag.width+(MAX_RADIUS*2));
+                nd[yORfy] = getBoundedCoord(nd[yORfy] + deltaY, tapestryDimensionsBeforeDrag.height+(MAX_RADIUS*2));
+            } else {
+                nd[xORfx] = getBoundedCoord(nd.x, tapestryDimensionsBeforeDrag.width);
+                nd[yORfy] = getBoundedCoord(nd.y, tapestryDimensionsBeforeDrag.height);
+            }
+        })
+    }
+
+    function dragended(d) {
+        if(!config.wpCanEditTapestry &&
+            tapestry.dataset.settings.nodeDraggable === false) {
+            return;
+        }
+
+        if (!d3.event.active) simulation.alphaTarget(0);
+
+        selection.forEach(nd => {
+            nd[xORfx] = nd.x;
+            nd[yORfy] = nd.y;
+            if (canEditNode(nd) && !autoLayout) {
+                $.ajax({
+                    url: config.apiUrl + "/tapestries/" + config.wpPostId + "/nodes/" + nd.id + "/coordinates",
+                    method: API_PUT_METHOD,
+                    data: JSON.stringify({x: nd.x, y: nd.y}),
+                    error: function(e) {
+                        alert("Sorry, there was an error saving the coordinates of this node!");
+                        console.error(e);
+                        //nd[xORfx] = nodesBeforeDrag.x;
+                        //nd[yORfy] = nodesBeforeDrag.y;
```

Comment:
Please do not leave commented out code without a comment why they are commented out. Please either remove or leave a comment as to why it's commented out.